DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/ViewDelegateControllerTests.cs

```diff
@@ -67,5 +67,41 @@ public void Deactivating_delegate_on_wrong_centre_returns_not_found_result()
             // then
             result.Should().BeNotFoundResult();
         }
+
+
+        [Test]
+        public void Reactivating_delegate_returns_redirect()
+        {
+            // given
+            A.CallTo(() => userDataService.GetDelegateUserCardById(1))
+                .Returns(new DelegateUserCard { CentreId = 2, Id = 1, Active = false});
+
+            // when
+            var result = viewDelegateController.ReactivateDelegate(1);
+
+            // then
+            result.Should().BeRedirectToActionResult();
+        }
+
+        [Test]
+        public void Reactivate_nonexistent_delegate_returns_not_found_result()
+        {
+            // when
+            var result = viewDelegateController.ReactivateDelegate(-1);

```

Comment:
For this test (and the wrong centre test) you should set up the call to userDataService.GetDelegateUserCardById return null (or a user with different centre) so that each test clearly does what it says in the name.

Reply:
I have used delegate id of different centre as suggested

Reply:
For clarity, the different delegate Ids won't do anything. The Ids are only used in the call to _userDataService.GetDelegateUserCardById_ which is mocked, so it does not actually make a call to the database. If you debug through the test you will see that the delegateUser details returned do not match the database. In order to get it to return the user, you will have to set up the mock call to return what you want, as you have done in  Reactivating_delegate_returns_redirect test.

Reply:
I have updated it already and I am not using id 10